### PR TITLE
regex-revalidate plugin update to prevent "restart" of all entries on each config file

### PR DESCRIPTION
**Pull request by chassell**
For issue #2651: Regex_Revalidate Plugin Existing Invalidations Re-Triggered Each Time a New Invalidation is Added

Keeps a sorted list of entries now to prevent duplicates and other minor updates.

### CODE REVIEW

File: plugins/regex_revalidate/regex_revalidate.c [hunk 1]
```diff
@@ -40,30 +40,19 @@
...
-{
-  return TSfree(s);
-}
-
 typedef struct invalidate_t {
   const char *regex_text;
   pcre *regex;
   pcre_extra *regex_extra;
-  time_t epoch;
+  time_t refresh;
```
Comment (chassell) [inline plugins/regex_revalidate/regex_revalidate.c]: as "epoch" is merely a time format... I think the purpose for the time is more intuitive.

File: plugins/regex_revalidate/regex_revalidate.c [hunk 2]
```diff
@@ -414,19 +496,20 @@ main_handler(TSCont cont, TSEvent event, void *edata)
...
         iptr   = pstate->invalidate_list;
-        while (iptr) {
+        while (iptr && iptr->expiry >= now) {
+          // only unexpired lines (from front)
           if (!date) {
             date = get_date_from_cached_hdr(txn);
             now  = time(NULL);
           }
-          if ((difftime(iptr->epoch, date) >= 0) && (difftime(iptr->expiry, now) >= 0)) {
+          if (iptr->refresh >= date) {
```
Comment (chassell) [inline plugins/regex_revalidate/regex_revalidate.c]: date is time_t so it must match a uint64_t if the system supports it. a double-conversion isn't needed.

File: plugins/regex_revalidate/regex_revalidate.c [hunk 3]
```diff
@@ -40,30 +40,19 @@
 #include <pcre.h>
 #endif
 
-#define LOG_PREFIX "regex_revalidate"
+#define PLUGIN_TAG "regex_revalidate"
```
Comment (chassell) [inline plugins/regex_revalidate/regex_revalidate.c]: changed only because all other plugins use this.

**Comment by PSUdaemon** [comment]
[approve ci]

**Comment by chassell** [comment]
-  if (oelt) { 
-    *pcutoff = oelt->expiry; 
+  if (oelt) {
+    *pcutoff = oelt->expiry;

trailing space.

**Pull request closed without merging**